src/reducers/authenticate.js

```diff
@@ -1,83 +1,46 @@
 const DEFAULT_STATE = {
-  loginJwt: {}
-}
-
-const logInWithJWT = (state, action) => {
-
-  let loginState = {}
-  let isAuthenticated = false
-
-  const id_token = action.payload.id_token
-  const access_token = action.payload.access_token
-  const username = action.payload.username
-  //TODO: expire token after the specified time
-  const expires_in = action.payload.expires_in
-
-  let time = new Date()
-  const expiry = time.setSeconds(time.getSeconds() + parseInt(expires_in))
-
-  if (id_token !== undefined && id_token !== state.loginJwt.id_token) {
-    loginState['id_token'] = id_token
-
-    if (access_token !== undefined && access_token !== state.loginJwt.access_token) {
-      loginState['access_token'] = access_token
-
-      if (username !== undefined && username !== state.loginJwt.username) {
-        loginState['username'] = username
-
-        if (expires_in !== undefined && expiry > new Date()) {
-          isAuthenticated = true
-        }
-      }
-    }
+  authenticationState: {
```

Comment:
i waffled on whether this was too much indirection, and whether the fields in this hash should just be at the top level of this subpart of the overall state.  the indirection is actually an artifact of me not realizing that the `combineReducers` function would do this sort of namespacing for us.  but once it was there, i decided to get other things working before revisiting this.

happy to do whatever people would prefer.

Reply:
working code wins!